### DISCUSSION TITLE
Issues to connect to the SignalK server With the ESP32 (SensESP)

### DISCUSSION THREAD

**Discussion by FDY-fr** [SignalK/SensESP#740] [Q&A]
I am writting this message as I am unable to establish a connection between my ESP32 fitted with SensESP (3.0.0-beta.1) and the SignalK server installed (from open plotter 2024-06-19-OpenPlotter-v4-Headless-stable) on my Raspberry PI 5.

I used platformIO to clone the SensESP-project-template example (https://github.com/SensESP/SensESP-project-template) on my PC. 
I compiled (built) it and uploaded it without any problem on my type C ESP32-WROOM-32D. after uploading the ESP32 built-in LED flashes indefinitely (200ms ON  -  200ms OFF). 

I am able to connect to he SensESP Access Point (see below screen copies) but I cannot configure my WiFi network as even if I press "Save" the network name and password is not saved. So I am guessing that my ESP32 is not connected to my wifi network and therefore cannot communicate to my SignalK server.  

Any help would be very much appreciated as I am a beginner on Arduino programming. Thanks in advance for your help. 

See below a copy of my PlatformIO serial monitor.  

Please build project in debug configuration to get more details about an exception.
See https://docs.platformio.org/page/projectconf/build_configurations.html

--- Terminal on COM5 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
D (44) configurable.cpp: Loading configuration for path '/system/hostname' from '/qLsPBGc6u2S+SPgr87i11VrDlCE='
D (59) configurable.cpp: Saving configuration path /system/hostname to file /qLsPBGc6u2S+SPgr87i11VrDlCE=
I (113) configurable.cpp: Could not find configuration for path /System/WiFi Settings
D (114) networking.cpp: Enabling Networking
D (136) Arduino: Arduino Event: 0 - WIFI_READY
V (809) Arduino: STA Started
V (812) Arduino: Configuring Station static IP: 0.0.0.0, MASK: 0.0.0.0, GW: 0.0.0.0
D (812) Arduino: Arduino Event: 2 - STA_START
V (827) Arduino: STA Stopped
D (828) Arduino: Arduino Event: 3 - STA_STOP
V (830) Arduino: AP Started
D (830) Arduino: Arduino Event: 10 - AP_START
I (830) networking.h: WiFi Access Point enabled, SSID: my-sensesp-project
I (837) networking.h: IP address of Device: 192.168.4.1
I (842) networking.cpp: Starting access point my-sensesp-project
I (879) configurable.cpp: Could not find configuration for path /system/httpserver
D (879) http_server.h: HTTPServer instance created
I (886) configurable.cpp: Not loading configuration: no config_path specified: 
I (887) configurable.cpp: Not loading configuration: no config_path specified:
D (897) configurable.cpp: Loading configuration for path '/System/Signal K Settings' from '/7NAVR03_FL3ag9_pt1mBQlpDmeM='
D (909) base_button.h: Button handler started
I (910) configurable.cpp: Not loading configuration: no config_path specified:
I (916) configurable.cpp: Not loading configuration: no config_path specified: 
I (954) configurable.cpp: Could not find configuration for path /Sensors/Analog Input/Voltage
I (984) configurable.cpp: Could not find configuration for path /Sensors/Digital Input 2/Value
D (985) main.cpp: Digital input value changed: 1
I (989) http_server.h: HTTP server started
E (991) Arduino: Failed adding service http.tcp.

D (995) signalk_ws_client.cpp: Starting SKWSClient
E (999) Arduino: Failed adding service signalk-sensesp.tcp.

I (999) signalk_ws_client.cpp: Initiating websocket connection with server...
D (1012) signalk_ws_client.cpp: Websocket is connecting to Signal K server on address 192.168.1.41:3000
D (1020) signalk_ws_client.cpp: No prior authorization token present.
I (1012) discovery.cpp: mDNS responder started for hostname 'my-sensesp-project'
D (1027) signalk_ws_client.cpp: Preparing a new access request
D (1040) signalk_ws_client.cpp: Access request: {"clientId":"08cf56ed-09b2-1763-9f17-1886105d9791","description":"SensESP device: my-sensesp-project","permissions":"readwrite"}
D (1055) signalk_ws_client.cpp: Access request url: http://192.168.1.41:3000/signalk/v1/access/requests
V (1064) Arduino: url: http://192.168.1.41:3000/signalk/v1/access/requests
D (1071) Arduino: protocol: http, host: 192.168.1.41 port: 3000 url: /signalk/v1/access/requests
D (1079) Arduino: request type: 'POST' redirCount: 0

D (1416) main.cpp: Analog input value: 0.142000
D (1916) main.cpp: Analog input value: 0.142000
D (2416) main.cpp: Analog input value: 0.142000
D (2916) main.cpp: Analog input value: 0.142000
D (3416) main.cpp: Analog input value: 0.142000
D (3916) main.cpp: Analog input value: 0.142000
D (4416) main.cpp: Analog input value: 0.142000
D (4916) main.cpp: Analog input value: 0.142000
D (5416) main.cpp: Analog input value: 0.142000
D (5916) main.cpp: Analog input value: 0.142000
I (6089) Arduino: select returned due to timeout 5000 ms for fd 52
D (6090) Arduino: failed connect to 192.168.1.41:3000
W (6090) Arduino: error(-1): connection refused
W (6093) Arduino: error(-4): not connected
D (6097) Arduino: tcp is closed

W (6100) signalk_ws_client.cpp: Can't handle response -1 to access request.
D (6107) signalk_ws_client.cpp:
I (6110) signalk_ws_client.cpp: Initiating websocket connection with server...
D (6117) signalk_ws_client.cpp: Websocket is connecting to Signal K server on address 192.168.1.41:3000
D (6126) signalk_ws_client.cpp: No prior authorization token present.
D (6132) signalk_ws_client.cpp: Preparing a new access request
D (6153) configurable.cpp: Saving configuration path /System/Signal K Settings to file /7NAVR03_FL3ag9_pt1mBQlpDmeM=
D (6165) signalk_ws_client.cpp: Access request: {"clientId":"21fd5f67-84f8-db65-03d0-f458233b86b5","description":"SensESP device: my-sensesp-project","permissions":"readwrite"}
D (6170) signalk_ws_client.cpp: Access request url: http://192.168.1.41:3000/signalk/v1/access/requests
V (6179) Arduino: url: http://192.168.1.41:3000/signalk/v1/access/requests
D (6186) Arduino: protocol: http, host: 192.168.1.41 port: 3000 url: /signalk/v1/access/requests
D (6195) Arduino: request type: 'POST' redirCount: 0

D (6416) main.cpp: Analog input value: 0.142000
D (6916) main.cpp: Analog input value: 0.142000
D (7416) main.cpp: Analog input value: 0.142000
D (7916) main.cpp: Analog input value: 0.142000
D (8416) main.cpp: Analog input value: 0.142000
D (8916) main.cpp: Analog input value: 0.142000
D (9416) main.cpp: Analog input value: 0.142000
D (9916) main.cpp: Analog input value: 0.142000
D (10416) main.cpp: Analog input value: 0.142000
D (10916) main.cpp: Analog input value: 0.142000
I (11202) Arduino: select returned due to timeout 5000 ms for fd 52
D (11203) Arduino: failed connect to 192.168.1.41:3000

<img width="1192" alt="image" src="https://github.com/user-attachments/assets/7171214f-dcc4-466d-807b-1b5290d534e8">
![IMG_5074](https://github.com/user-attachments/assets/334776dc-3f91-4d2e-a0af-404d4874c604)

**Comment by joelkoz** [comment]
According to your logs, your device IS connecting to your server:

> I (999) signalk_ws_client.cpp: Initiating websocket connection with server...
> D (1012) signalk_ws_client.cpp: Websocket is connecting to Signal K server on address 192.168.1.41:3000
> D (1020) signalk_ws_client.cpp: No prior authorization token present.

“No prior authorization token present” indicates you haven’t performed step #2, which is going to the server and authorizing the connection. When you log in to the server, you should see a pending access request from your device. Once you give it the green light, your device should show up.

**Reply by FDY-fr** [reply]
Thanks Joel for your reply.  
i found the solution on the “discord.com” in the SensESP community.   The problem was that I was not connected to my WiFi network.  
I fixed my problem by changing Platformio.ini to.     lib_deps = https://github.com/SignalK/SensESP.git which has a more up-to-date version on the SensESP  libraries 
Thanks
